src/styles/_catalog.scss

```diff
@@ -1,22 +1,30 @@
+$card-width: 200px;
+
 .catalog {
-  display: flex;
-  flex-wrap: wrap;
+  display: grid;
+  justify-content: center;
 
-  max-width: 944px;
-  margin: 0 auto;
+  grid-template-columns: repeat(4, $card-width);
+  gap: 48px;
 
   &__card {
-    display: flex;
-    justify-content: center;
-    align-items: center;
-    width: 200px;
+    display: grid;
+    place-content: center;
+
     height: 300px;
     border: 1px solid #ccc;
     border-radius: 5px;
-    margin-bottom: 48px;
   }
 
-  &__card:not(:nth-child(4n)) {
-    margin-right: 48px;
+  @media (max-width: 1023px) {
+    grid-template-columns: repeat(3, $card-width);
+  }
+
+  @media (max-width: 767px) {
+    grid-template-columns: repeat(2, $card-width);
+  }
+
+  @media (max-width: 487px) {
+    grid-template-columns: $card-width;
   }
```

Comment:
You can make a loop to create all media queries. Just create a correctly mapped list to iterate.